### ISSUE TITLE
Think about metadata from the start

### ISSUE THREAD

**Issue by libcce** [gvwilson/10-dm#11]
CodeMeta - https://codemeta.github.io/
A CodeMeta instance file describes the metadata associated with a software object using JSON's linked data (JSON-LD) notation. The goal of CodeMeta is to create a concept vocabulary that can be used to standardize the exchange of software metadata across repositories and organizations.
Tools to generate CodeMeta - https://codemeta.github.io/tools/
CodeMeta example - https://github.com/ropensci/codemetar/blob/master/codemeta.json
CodeMeta can live within a GitHub (or other) project repository until deposited in a repository
According to Software Heritage Library, other sources for metadata: package management file, CITATION file, .About file - https://www.rd-alliance.org/sites/default/files/attachment/2017-09-19-RDA-IG.pdf

Many repositories moving to support schema.org + recent announcement of Google Dataset Search which relies on schema.org - https://www.blog.google/products/search/making-it-easier-discover-datasets/
Move to structured metadata which can be used by search engines, repositories... guidelines found here - https://developers.google.com/search/docs/data-types/dataset w/ examples and tools
Researchers can also see what metadata they will need for repositories (test the waters) by using sandbox.zenodo.org
To aid in linking your research objects and ultimately the discovery of your research, ORCID and related identifiers are important - https://authorcarpentry.github.io/dois-citation-data/01-register-doi.html (also linking to the funder and determining the license)
Linking your research early helps you get credit for all your research and helps aid publication/discovery/reproducibility
Tools available to determine the best license to use: Creative Commons Wizard, Choose a License, TLDR Legal, etc

Will be helpful to include paper-data-software examples (to critique approaches)
TIB Hannover FAIR Data & Software used Pangea paper as good example:
Koen Kole, Rik G.H. Lindeboom, Marijke P.A. Baltissen, Pascal W.T.C. Jansen, Michiel Vermeulen, Paul Tiesinga, Tansu Celikel (2017): Proteomic landscape of the primary somatosensory cortex upon sensory deprivation, GigaScience, Volume 6, Issue 10, 1 October 2017, Pages 1–10. DOI https://doi.org/10.1093/gigascience/gix082

**Comment by gvwilson** [comment]
- :+1: to mentioning ORCIDs and DOIs (I think they're separable from the other ideas in this rule).
- If the focus of this paper is "things you can actually *do*, right now", then telling people to "think about" CodeMeta and schema.org may not be a fit - we need to show them what to create and how, and where to put it. Can we work through an example in the scope of this paper? And will that example be enough to get people rolling with their own data? (I've never used JSON-LD or the like...)

**Comment by libcce** [comment]
@gvwilson I agree. What is probably more realistic is focusing on DataCite metadata, explaining the benefits of PIDs (ORCID, DOIs), related identifiers and linking your research works, adding the appropriate license, etc. I believe sandbox.zenodo.org is a great tool for seeing how the data, software, etc. you deposit will looking like when you submit and publish it. There might be a caveat here that authors should also check with the publishers/journals to see if they are linking to their research works. I'm not sure if we want to mention Schema/JSON-LD but Google's recent push into improving search for structured data (Google Dataset Search) might be a reason for noting it for future use.
